mario/tools/parsersclass.py

```diff
@@ -99,6 +100,10 @@ def parse_from_txt(
 
     matrices, indeces, units = txt_parser(path, table, mode, sep)
 
+    nan_keys = [key for key, df in matrices['baseline'].items() if isinstance(df, pd.DataFrame) and df.isna().any().any()]
+    if nan_keys:
+        raise ValueError(f"NaN values found in the following matrices: {nan_keys}")
```

Comment:
like it a lot! Was missed.

Although the way that is implemented is not the correct and most efficient ones. My suggestion is to do it when passing the data to the class initiator in Core module. In thins way, you can do it only once for all situations ,regardless of the parser you are using

Reply:
done, I moved it to core model, used only in the "if init_with_parser" clause